app-frontend/src/app/pages/projects/edit/annotate/annotate.controller.js

```diff
@@ -166,13 +163,20 @@ export default class AnnotateController {
                 combo: 'e',
                 description: 'Edit annotation',
                 callback: () => {
-                    if (this.clickedId) {
-                        this.onUpdateAnnotationStart(
-                            _.filter(this.annoToExport.features, (f) => {
-                                return f.properties.id === this.clickedId;
-                            })[0]
-                        );
-                    }
+                    this.getMap().then((mapWrapper) => {
+                        let drawLayer = mapWrapper.getLayers('draw')[0];
+                        if (!_.isEmpty(drawLayer) && drawLayer.transform) {
+                            this.disableTransformHandler(mapWrapper);
+                            let drawLayerGeometry = drawLayer.toGeoJSON().geometry;
+                            this.createEditableDrawLayer(mapWrapper, drawLayerGeometry);
+                        } else if (this.clickedId) {
+                            this.onUpdateAnnotationStart(
+                                _.filter(this.annoToExport.features, (f) => {
```

Comment:
If you're taking the first element of the filtered array, you can also use es6 `find`, which will work similarly but stop after finding the first matching element.